src/app/layout/header/header.component.ts

```diff
@@ -30,18 +31,23 @@ export class HeaderComponent implements OnInit, OnDestroy {
 
   menuCallbacks: Map<String, MenuHiddenCallback> = new Map<String, MenuHiddenCallback>([
     [
-      '_settings', function(headerComponent) {
+      '_settings', function(headerComponent, menuItem) {
         return headerComponent.checkContextUserEqualsLoggedInUser();
       }
     ],
     [
-      '_resources', function(headerComponent) {
+      '_resources', function(headerComponent, menuItem) {
         return headerComponent.checkContextUserEqualsLoggedInUser();
       }
     ],
     [
-      'settings', function(headerComponent) {
-        return headerComponent.loggedInUserNotSpaceAdmin();
+      'settings', function(headerComponent, menuItem) {
+        const subFeature = menuItem['subFeature'];
+        const allFeatures = headerComponent.context.user['features'];
+        if (headerComponent.menusService.isFeatureUserEnabled(subFeature, allFeatures)) {
```

Comment:
@rohitkrai03 please add a check for `subFeature` and `allFeatures` as these can be undefined.